### PR TITLE
Enhance unit tests #12

### PR DESCRIPTION
**Pull request by gentilboulet**
We've lost 100% coverage on reducer.
This first PR is half of the work on this issue.

Missing tests for reducers : cost, some bonuses, discounts
Missing tests for containers.

### CODE REVIEW

**Comment by CyrilleHugues** [comment]
image
you are a terrorist.

File: src/actions/header.tsx [hunk 1]
```diff
@@ -27,28 +30,30 @@ export interface IHeaderSetRank {
 
 export type IHeaderAction = IHeaderSetName | IHeaderSetConcept | IHeaderSetArchetype | IHeaderSetRank;
 
-export function headerSetName(s: string): IHeaderSetName {
+export function setName(s: string): IHeaderSetName {
```
Comment (CyrilleHugues) [inline src/actions/header.tsx]: this is nice

File: src/actions/header.tsx [hunk 2]
```diff
@@ -1,7 +1,10 @@
 import * as constants from '../constants/header';
 import { IPerk } from '../types/perks';
-import { IDataRank } from '../types/ranks';
+
```
Comment (CyrilleHugues) [inline src/actions/header.tsx]: why ?

File: src/reducers/header.test.tsx [hunk 3]
```diff
@@ -8,31 +8,51 @@ import { globalReducer } from './global';
 
 const initialState: IStoreState  = initialStateFactory();
 
-it('HEADER_SET_NAME', () => {
-  const refState = headerReducer(initialState, actions.headerSetName('Robert'));
-  expect( refState ).toMatchSnapshot();
-  expect( globalReducer(initialState, actions.headerSetName('Robert') ) ).toMatchObject(refState);
-});
+describe('Testing headerReducer', () => {
+  it('HEADER_SET_NAME', () => {
```
Comment (CyrilleHugues) [inline src/reducers/header.test.tsx]: good form but your `it` still lacks verb :p

File: src/reducers/global.test.tsx [hunk 4]
```diff
@@ -6,8 +6,10 @@ import { globalReducer } from './global';
 
 const initialState: IStoreState  = initialStateFactory();
 
-it('JUNK', () => {
-  expect( initialState ).toMatchSnapshot();
-  const junk = { type: 'JUNK_ACTION' };
-  expect( globalReducer(initialState, junk as IAction )).toMatchObject( initialState );
+describe('Testing globalReducer', () => {
+  it('Sending junk action', () => {
```
Comment (CyrilleHugues) [inline src/reducers/global.test.tsx]: `should send junk action` as it is a test checking it really happens

File: src/actions/header.tsx (hunk 2 above)
Comment (gentilboulet) [inline src/actions/header.tsx]: Moved data lower, in a similar pattern for ranks & archetypes (data then type).

File: src/reducers/costs.tsx [hunk 5]
```diff
@@ -4,6 +4,7 @@ import { ICost } from '../types/costs';
 import { updateReductions } from './reductions';
 
 export function applyCost(oldState: IStoreState, cost: ICost): IStoreState {
+  if (typeof cost === 'undefined') { throw 'Internal error - undefined cost'; }
```
Comment (CyrilleHugues) [inline src/reducers/costs.tsx]: Doesn't it already throw a TypeError ?

Comment (gentilboulet) [inline src/reducers/costs.tsx]: I don't know why right now, but with the test script, tslint did not warn.
Can be redundant.

File: src/actions/header.tsx (hunk 2 above)
Comment (CyrilleHugues) [inline src/actions/header.tsx]: ok, you might want to make this standard on import order cristal clear in documentation.

1. node modules deps
2. interface types
3. services followed by their types

Is that correct ?
Do you have to leave an empty line between 2 and 3 ?

File: src/reducers/bonuses.test.tsx [hunk 6]
```diff
@@ -6,14 +6,51 @@ import { applyBonuses } from './bonuses';
...
+  });
+
+  it('should receive a BONUS_SKILL_RANK action', () => {
+    const bonuses = [actions.skillRank('Awareness')];
+    const state = applyBonuses(initialState, bonuses);
+    expect( initialState.getIn(['skills', 0, 'value']) ).toBe(0);
+    expect( state.getIn(['skills', 0, 'value']) ).toBe(5);
+  });
+
+  /*it('BONUS_SPECIALITY', () => {
```
Comment (CyrilleHugues) [inline src/reducers/bonuses.test.tsx]: To be deleted

File: src/reducers/header.test.tsx [hunk 7]
```diff
@@ -8,31 +8,51 @@ import { globalReducer } from './global';
 
 const initialState: IStoreState  = initialStateFactory();
 
-it('HEADER_SET_NAME', () => {
-  const refState = headerReducer(initialState, actions.headerSetName('Robert'));
-  expect( refState ).toMatchSnapshot();
-  expect( globalReducer(initialState, actions.headerSetName('Robert') ) ).toMatchObject(refState);
-});
+describe('Testing headerReducer', () => {
+  it('shoudl receive a HEADER_SET_NAME action', () => {
```
Comment (CyrilleHugues) [inline src/reducers/header.test.tsx]: Should, error is on every test

Comment (gentilboulet) [inline src/reducers/header.test.tsx]: butterfingers strike again.

**Pull request merged**